spawn/spawn.go

```diff
@@ -1,25 +1,45 @@
 package spawn
 
 import (
+	"context"
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 	"time"
 
+	"github.com/evergreen-ci/evergreen"
 	"github.com/evergreen-ci/evergreen/cloud"
+	"github.com/evergreen-ci/evergreen/cloud/providers"
 	"github.com/evergreen-ci/evergreen/cloud/providers/ec2"
 	"github.com/evergreen-ci/evergreen/model/distro"
 	"github.com/evergreen-ci/evergreen/model/host"
 	"github.com/evergreen-ci/evergreen/model/user"
+	"github.com/evergreen-ci/evergreen/subprocess"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/send"
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v2"
 )
 
+// each regex matches one of the 5 categories listed here:
+// https://technet.microsoft.com/en-us/library/cc786468(v=ws.10).aspx
+var passwordRegexps = []*regexp.Regexp{
+	regexp.MustCompile(`[\p{Ll}]`), // lowercase letter
+	regexp.MustCompile(`[\p{Lu}]`), // uppercase letter
+	regexp.MustCompile(`[0-9]`),
+	regexp.MustCompile(`[~!@#$%^&*_\-+=|\\\(\){}\[\]:;"'<>,.?/` + "`]"),
+	regexp.MustCompile(`[\p{Lo}]`), // letters without upper/lower variants (ex: Japanese)
+}
+
 const (
-	MaxPerUser        = 3
-	DefaultExpiration = time.Duration(24 * time.Hour)
+	MaxPerUser                 = 3
+	DefaultExpiration          = 24 * time.Hour
+	MaxExpirationDurationHours = 24 * 7 // 7 days
```

Comment:
I'd try and keep duration info in time.Duration values rather than ints

(it's also ~days because it doesn't account for leapseconds)